File: labo1/exercise4/Doublon.py
import random

def est_sans_doublon(T):
    n = len(T)
    T.sort()
    for i in range (n - 1):
        if T[i] == T[i + 1]:
            return False
    return True


tab = []
n = int(input("Veuillez entrer la taille du tableau souhaitée : "))
for i in range(n):
    tab.append(random.randint(0,100))

print(tab)

contient_des_doublons = est_sans_doublon(tab)
if contient_des_doublons:
    print("Tableau sans doublon")
else:
    print("Tableau avec doublon")

File: labo1/exercise4/Doublon.ml
let rec estSansDoublon tab =
  match tab with
  | [] -> true
  | [_] -> true
  | x :: y :: xs when x = y -> false
  | _ :: _ :: xs -> estSansDoublon xs


let estSansDoublonImperatif tab = 
  let n = Array.length tab - 1
  in
  Array.sort (Int.compare) tab;
  try 
    for i = 0 to n do
      if i <> n && tab.(i) = tab.(i + 1) then 
        raise Exit
    done;
    true
  with
  | Exit -> false
